display/displaytest/text.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2024 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package displaytest
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"periph.io/x/conn/v3/display"
+)
+
+// TestTextDisplay exercises the methods provided by the interface. It can be
+// used interactively as a quick smoke test of an implementation, and from test
+// routines. This doesn't test brightness or contrast to avoid EEPROM wear
+// issues.
+func TestTextDisplay(dev display.TextDisplay, interactive bool) []error {
+	result := make([]error, 0)
+	var err error
+
+	pauseTime := time.Millisecond
```

Comment:
I'd vote for not sleep at all when not interactive. These 1ms can add up quickly over time. Making it an argument would probably be nice, then: `interfactive := pauseTime != 0`

Reply:
I would rather burn 1ms than have two additional lines per sleep call:

I've changed it to 0, which returns immediately. I'd rather burn the extra function call than have conditionals around every one.